### ISSUE TITLE
[2.0.5][2.0.11][Urgent] OException: Cannot find best cluster for class 

### ISSUE THREAD

**Issue by mesonoxian** [orientechnologies/orientdb#4835]
Orientdb version - 2.0.11

In distributed mode when **schema change** is made, it does not replicate the schema change to other nodes and therefore when we try to perform a write on  the other node it throws exception

```
 com.orientechnologies.common.exception.OException: Cannot find best cluster for class 'RManagement' on server 'node1435865189367'. ClusterStrategy=local
com.orientechnologies.common.exception.OException: Cannot find best cluster for class 'RManagement' on server 'node1435865189367'. ClusterStrategy=local
	at com.orientechnologies.orient.server.distributed.OLocalClusterStrategy.readConfiguration(OLocalClusterStrategy.java:90)
	at com.orientechnologies.orient.server.distributed.OLocalClusterStrategy.getCluster(OLocalClusterStrategy.java:60)
	at com.orientechnologies.orient.core.metadata.schema.OImmutableClass.getClusterForNewInstance(OImmutableClass.java:273)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.save(ODatabaseDocumentTx.java:2272)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.save(ODatabaseDocumentTx.java:118)
	at com.orientechnologies.orient.core.record.impl.ODocument.save(ODocument.java:1727)
	at com.orientechnologies.orient.core.record.impl.ODocument.save(ODocument.java:1718)
	at com.tinkerpop.blueprints.impls.orient.OrientElement.save(OrientElement.java:311)
	at com.tinkerpop.blueprints.impls.orient.OrientElement.save(OrientElement.java:291)
	at com.orientechnologies.orient.graph.sql.OCommandExecutorSQLCreateVertex$1.call(OCommandExecutorSQLCreateVertex.java:126)
	at com.orientechnologies.orient.graph.sql.OCommandExecutorSQLCreateVertex$1.call(OCommandExecutorSQLCreateVertex.java:106)
	at com.orientechnologies.orient.graph.sql.OGraphCommandExecutorSQLFactory.runInTx(OGraphCommandExecutorSQLFactory.java:123)
	at com.orientechnologies.orient.graph.sql.OGraphCommandExecutorSQLFactory.runInTx(OGraphCommandExecutorSQLFactory.java:142)
	at com.orientechnologies.orient.graph.sql.OCommandExecutorSQLCreateVertex.execute(OCommandExecutorSQLCreateVertex.java:106)
	at com.orientechnologies.orient.core.sql.OCommandExecutorSQLDelegate.execute(OCommandExecutorSQLDelegate.java:64)
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.executeCommand(OAbstractPaginatedStorage.java:1194)
	at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.command(OAbstractPaginatedStorage.java:1183)
	at com.orientechnologies.orient.server.distributed.ODistributedStorage.command(ODistributedStorage.java:250)
	at com.orientechnologies.orient.core.command.OCommandRequestTextAbstract.execute(OCommandRequestTextAbstract.java:63)
	at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.command(ONetworkProtocolBinary.java:1179)
	at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:386)
	at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.execute(OBinaryNetworkProtocolAbstract.java:217)
	at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:69)

```

Is there any **intermediate** fix for this so that we can get rid of this problem in our Prod like environment. Upgrade and any of such activities is not going to be an option and as there is some overhead in getting access to them and moving data etc.

Any help regarding the same will be appreciated

Pairing,
@mesonoxian / @machinelearner

**Comment by lvca** [comment]
Please upgrade to last stable 2.0.14.

**Comment by machinelearner** [comment]
Hey Luca, sorry, this problem still exists in this version as well.

We also noticed another thing, the client does not get updated with the schema and partition change. We have to bounce the application for the client to pick up these changes. I have mentioned this in prjhub before.

**Comment by lvca** [comment]
Guys, how can I reproduce it? Are you stuck on 2.0 or do you have any chance to test it against 2.1?

**Comment by machinelearner** [comment]
Sorry @lvca been a long and busy week with major release.

We have tested this on 2.0.5, 2.0.11, 2.0.14

Here is a simple way to reproduce this.
* In distributed mode(3 node default config), create a custom vertex definition.

   ``` Create class Role extends V ;Create property Role.createdOn Datetime;Create property Role.modifiedOn Datetime;Create property Role.statusChangedOn Datetime;```
*  Create a instance of a vertex

  ``` Create vertex Role content some_json ```

Two set of problems you will observe:

1.  In the client where it says unable to locate the schema(the client was not able to get this schema propagation)
2.  If a request goes to a node which was not responsible for creation of schema, it throws the above exception.

we have seen a bunch of other things(not related to this). I will send a separate mail to support email with threadDump and heap dump.

**Comment by lvca** [comment]
@machinelearner Did you have the chance to test the same against 2.1.1?

**Comment by machinelearner** [comment]
@lvca not really, ODB is just one of the many moving pieces. If this was the only piece being developed, would've perhaps had the luxury to switch between things quickly. I hope you understand.

**Comment by lvca** [comment]
@machinelearner do you still have this problem? I've just fixed it in 2.1.x

**Comment by lvca** [comment]
So I'm closing it. If the problem persists, please comment this issue again (don't open a new issue). Thanks.
